RentACarWebAPI/Controllers/CarController.cs

```diff
@@ -58,15 +58,15 @@ public async Task<IActionResult> GetAsync(int id)
         // POST api/<CarController>
         [HttpPost]
         [ProducesResponseType(StatusCodes.Status201Created, Type = typeof(CarDto))]
-        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
         public async Task<IActionResult> PostAsync([FromBody] CarDto carDto)
         {
             var carEntity = carDto.ToCarEntity(carDto);
 
             var carCreated = await _carService.CreateAsync(carEntity);
 
             if (carCreated is null)
-                return NotFound();
+                return BadRequest();
```

Comment:
hmm, tricky question: why is this a bad request? not saying the change is bad, just wondering why you changed it

Reply:
I think that if the `Post` fails is more probable that the object doesn't comply some requirement or business rule than something `not found`. Honesty, I'm not sure, but it sounds better to me.

Reply:
a bad request response implies there's something terrible in the request sent from the client, and here what potentially failed is the insert to the DB and _NOT_ the post by itself. I believe the only change `carCreated` would be null is if an exception occurred, and you won't be getting it there

Reply:
Ohhh okey, understood! I'll put it back like it was before. Thanks for the explanation and for your time!

Reply:
no, what was before was "worse" than this. a bad request here would have sense if your service performs some kind of validation and returns null if something isn't right.. if I were you I would return the created object without any null check